### PR TITLE
Miscellaneous refactoring, continued

### PR DESCRIPTION
**Pull request by sestephens73**
I (Sean) had to merge #112 early because it was blocking me. This is a continuation of #112 addressing comments and concerns there.

### CODE REVIEW

File: parla/task_runtime.py [hunk 1]
```diff
@@ -428,84 +422,84 @@ def is_assigned(self):
...
-    def predecessors(self, predecessors: Collection['Task']):
-        self._predecessors: List[Task] = list(predecessors)
-        self._num_blocking_predecessors: int = len(predecessors)
-        for dep in predecessors:
+    def dependencies(self) -> Tuple["Task"]:
+        return tuple(self._dependencies)
+
+    @dependencies.setter
+    def dependencies(self, dependencies: Collection['Task']):
+        self._dependencies: List[Task] = list(dependencies)
```
Comment (dialecticDolt) [inline parla/task_runtime.py:430]: Why is dependencies a collection and not just a list from the start?

File: parla/task_runtime.py [hunk 2]
```diff
@@ -428,84 +422,84 @@ def is_assigned(self):
...
         with self._mutex:
-            self._num_blocking_predecessors -= 1
+            self._num_blocking_dependencies -= 1
             # Add events from one dependent task.
             # (We are aiming to multiple device tasks, and it would
             #  be possible to have multiple events)
             if events is not None:
-                self._predecessor_events.append(events)
+                self._dependency_events.append(events)
             self._maybe_ready_to_schedule()
```
Comment (dialecticDolt) [inline parla/task_runtime.py:500]: There was a comment on the old PR about this needed a better name, just putting it here again.

**Comment by sestephens73** [comment]
All the (major) comments from #112 should be addressed at this point.

Tested on matmul: `python matmul_automatic.py 1`

File: parla/task_runtime.py (hunk 1 above)
Comment (nicelhc13) [reply parla/task_runtime.py:430]: Same question; this exists from 2yr olds.

Comment (dialecticDolt) [reply parla/task_runtime.py:430]: Also less importantly this type hint is not strictly accurate, as this Collection can also have taskIDs (as mentioned in the comment near there)

**Review by nicelhc13 — APPROVED** [review]
lgtm

**Pull request merged**